src/shared/edit-util.c

```diff
@@ -291,6 +300,10 @@ static int run_editor(const EditFileContext *context) {
         }
 
         return 0;
+
+oom:
+        log_oom();
+        _exit(EXIT_FAILURE);
```

Comment:
Please move this to the child block.

I think it is better to move the child operation into a separated function:
```c
r = safe_fork(...);
if (r < 0)
        return r;
if (r == 0) {
        r = run_editor_child(...);
        _exit(r < 0 ? EXIT_FAILURE : EXIT_SUCCESS);
}
return 0;
```

Then, you can shorten the error handling in `run_editor_child()` something like the following:
```c
r = strv_extend(...);
if (r < 0)
        return log_oom();
```